packages/typespec-ts/test/modularIntegration/generated/client/structure/default/src/api/operations.ts

```diff
@@ -34,7 +35,13 @@ export function _oneSend(
 
 export async function _oneDeserialize(result: One204Response): Promise<void> {
   if (result.status !== "204") {
-    throw result.body;
+    const internalError = (result.body as any).error || result.body || result;
+    const message = `Unexpected status code ${result.status}`;
+    throw new RestError(internalError.message ?? message, {
+      statusCode: result.status,
+      request: result.request,
+      response: result.body as PipelineResponse,
+    });
```

Comment:
@deyaaeldeen as HttpResponse doesn't have a response property and I convert the body into the PipelineResponse here, could you help provide some insights ? Thank you.